src/patternfly/components/Form/styles.scss

```diff
@@ -227,7 +228,8 @@ $pf-c-form__select--m-error-svg--Viewbox: "0 0 26 15.8" !default;
     line-height: var(--pf-c-form__input--LineHeight);
     color: var(--pf-c-form__input--Color);
     background-color: var(--pf-c-form__input--BackgroundColor);
-    border: 0;
+    border: var(--pf-c-form__input--BorderSize) solid;
+    border-color: var(--pf-c-form__input--BorderColor--transparent);
```

Comment:
You had this comment in the issue:

> So I've noticed that when adding border: 1px solid transparent; to the dropdown menus, it actually applies a solid border to the element. It looks as if it is picking up the box-shadow in some way?
>
> The inputs and buttons work as expected - no border appears, unless in High Contrast mode.

I actually see this issue for inputs as well. 

Try moving the border-color into `border` like this:
`border: var(--pf-c-form__input--BorderSize) solid var(--pf-c-form__input--BorderColor--transparent);`

(what's weird is that if I replaced the variable with "transparent" in the css as you have it, that works too)

Related to inputs, I'm seeing that the heights arent' exactly the same between inputs and buttons: 

![image](https://user-images.githubusercontent.com/21063328/41317563-8e194002-6e63-11e8-950e-732c226ef0b5.png)

Reply:
I added the transparent border to the primary, secondary and tertiary buttons as well, which fixed the height issue. I guess adding it to `pf-c-button` wasn't enough.

Reply:
I also changed the `border` property to be `border: var(**--BorderWidth) solid var(**--BorderColor--transparent);` and that eliminated the extra `1px` border that was added :)

Reply:
fantastic, thanks Adam